Topic pills redesign

oetherington opened the pull request:
This PR adds the new topic pill styles:
<img width="712" alt="Screenshot 2023-03-07 at 11 12 04" src="https://user-images.githubusercontent.com/5075628/223406725-bc0e2ef5-bfff-4fc0-819d-329116dc1de9.png">
<img width="679" alt="Screenshot 2023-03-07 at 11 12 31" src="https://user-images.githubusercontent.com/5075628/223406742-e835c4f7-bfdd-42c4-b8ab-87d797afaac7.png">

And hovered:
<img width="133" alt="Screenshot 2023-03-07 at 11 12 11" src="https://user-images.githubusercontent.com/5075628/223406764-bc7226cb-589b-4433-b60d-c94efb00b5c5.png">
<img width="198" alt="Screenshot 2023-03-07 at 11 12 19" src="https://user-images.githubusercontent.com/5075628/223406770-e87099e4-0ce3-4a5c-becd-6d7eae611b6c.png">
<img width="141" alt="Screenshot 2023-03-07 at 11 12 24" src="https://user-images.githubusercontent.com/5075628/223406782-d5e7ae60-574a-4f9f-bd52-2adf296cbecf.png">

I also updated the close button as I saw this was on the todo list and it was trivial to add here:
<img width="221" alt="Screenshot 2023-03-07 at 11 16 00" src="https://user-images.githubusercontent.com/5075628/223407124-7fcb2e7a-9ca3-49cf-a94a-edd89be0cfc7.png">

Two open questions:

- Do we need to adjust the sizing of the icons. In the first screenshot I posted the "Wild Animal Welfare" icon looks about the right size to me, but "Biosecurity" may be a little to big. Doing this is a little fiddly, so I figured I'd get another opinion before wasting a load of time on it.
- Do we want to make any changes to the core topics section on the new posts page? We could just add the new core topic styles, but we'd need some way to distinguish between a selected core tag and an unselected core tag:
<img width="538" alt="Screenshot 2023-03-07 at 11 18 19" src="https://user-images.githubusercontent.com/5075628/223407982-3119fc93-5a47-49f6-97f6-4d01fd98ac90.png">



┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1204123390082356) by [Unito](https://www.unito.io)

agnesstenlund commented:
nice one, this looks really great 👏 👏  

> Do we need to adjust the sizing of the icons. In the first screenshot I posted the "Wild Animal Welfare" icon looks about the right size to me, but "Biosecurity" may be a little to big. Doing this is a little fiddly, so I figured I'd get another opinion before wasting a load of time on it.

yup I think we need to do that. May be better for the svgs to be the same size though instead of you tweaking the code for each one? I could export and update them if you think that'd be better

> Do we want to make any changes to the core topics section on the new posts page? We could just add the new core topic styles, but we'd need some way to distinguish between a selected core tag and an unselected core tag:

good question! If it's not too much work I would go for new core topic styles with a 60% opacity for those not selected

I would also either decrease the height of the pills _or_ increase the width (usually there is a bit more padding on the sides than top/bottom–in the designs I have 4px top/bottom and 8px left/right)

Will-Howard commented on packages/lesswrong/components/posts/PostsPage/PostsPagePostHeader.tsx:
```diff
@@ -192,8 +192,6 @@ const PostsPagePostHeader = ({post, answers = [], toggleEmbeddedPlayer, hideMenu
     <AnalyticsContext pageSectionContext="topSequenceNavigation">
       <PostsTopSequencesNav post={post} />
     </AnalyticsContext>
-    {!post.group && !post.sequence && !post.question && <PostsPageTopTag post={post} />}
```
Are we just getting rid of this? Was this in the designs or did you just infer it?

oetherington replied:
Yep - it's in the designs 🙂

Will-Howard reviewed (approved):
Some of the icons look a bit close to the left border to me, maybe add a bit of padding?
![Screenshot 2023-03-08 at 11 12 29](https://user-images.githubusercontent.com/77623106/223699805-8fe067b7-327f-4ab3-b28b-b4f0be371538.png)

Otherwise this looks great, approving subject to design review

oetherington commented:
@agnesstenlund Great! If it's quick and easy to reexport the svgs in the same size then that's probably the best long-term solution, but let me know if it's too much work.

I'll go ahead and add the extra padding and unselected styles.

oetherington commented:
@agnesstenlund I've just added the updated icons (thanks again for re-exporting!):
<img width="591" alt="Screenshot 2023-03-09 at 18 31 18" src="https://user-images.githubusercontent.com/5075628/224122159-d7c810e1-160d-469a-a7c3-581046649c25.png">
<img width="614" alt="Screenshot 2023-03-09 at 18 31 14" src="https://user-images.githubusercontent.com/5075628/224122168-ac73c58a-5e8a-4144-b742-02c96ddf6b80.png">

I also added a little bit of padding on the icon, but I wouldn't be surprised if that needs a little bit of tweaking.

For testing, I also created the post [http://localhost:3000/posts/oe7kFXbbvmFFsnqgM/test-locking-comments](http://localhost:3000/posts/oe7kFXbbvmFFsnqgM/test-locking-comments) in the dev database which has all of the core tags applied to it.

Here are the new styles for selected/unselected core tags in the new post page options:
<img width="545" alt="Screenshot 2023-03-09 at 18 35 38" src="https://user-images.githubusercontent.com/5075628/224122864-8b7dffe7-cd12-471a-af66-192187536b6b.png">
<img width="543" alt="Screenshot 2023-03-09 at 18 35 32" src="https://user-images.githubusercontent.com/5075628/224122868-af0cb097-4d64-42fe-877c-50ae73bf0cdb.png">

agnesstenlund commented:
really nice! seeing this there are still a couple ones I'd like to tweak that I missed–will send those over 👍

agnesstenlund commented:
Also, I think we just have "Animal welfare" as a core topic, not farmed and wild? 🤔 At least according to the topics page 
![Screenshot 2023-03-10 at 11 56 47](https://user-images.githubusercontent.com/13235378/224298710-dbc90960-d8e3-4d29-8240-6eebf35d1798.png)

agnesstenlund commented:
here they are! I doubt they have the right names so I added this image to show which one should go where:
![Screenshot 2023-03-10 at 13 31 00](https://user-images.githubusercontent.com/13235378/224316715-8f4bf801-26c4-4735-be2e-25b2f6e5ccdb.png)

SVGs below 👇 
![aiSafetyIcon](https://user-images.githubusercontent.com/13235378/224317296-26f74d94-f729-47b9-92a4-949dc7f5033e.svg)
![causePrioritisationIcon](https://user-images.githubusercontent.com/13235378/224318022-7edb16a6-820a-4fc7-8139-5e67d8ded81f.svg)
![communityGlobeIcon](https://user-images.githubusercontent.com/13235378/224317300-62c3d3c4-761b-44a6-b8f8-57ec26e84a35.svg)
![groupsIcon](https://user-images.githubusercontent.com/13235378/224317302-bb955049-1dd9-4d39-bbda-908da8420f0b.svg)
![policyIcon](https://user-images.githubusercontent.com/13235378/224317304-7847d19b-42bb-4d33-9d79-a4d5c7092616.svg)

oetherington commented:
@agnesstenlund Great - thanks! I've just added the new icons:
<img width="630" alt="Screenshot 2023-03-13 at 11 49 55" src="https://user-images.githubusercontent.com/5075628/224694003-b28b99c8-2455-4397-b55f-a02d493476da.png">
<img width="623" alt="Screenshot 2023-03-13 at 11 49 44" src="https://user-images.githubusercontent.com/5075628/224694021-9ddc2706-38d9-4f65-88ce-2ae332d7462c.png">

Let me know if you think they need any tweaks, otherwise I think this branch is done.

Also, it looks like you're correct about it just being "Animal welfare" rather than separate tags for "wild" and "farmed" - it's just old data in the dev database. On prod, animal welfare uses the same icon as wild animal welfare in the screenshots above.

agnesstenlund commented:
looks great! 🚢
